### PR TITLE
Rchain 3132: modify DeployData structure

### PR DESCRIPTION
**Pull request by odeac**
## Overview
- user renamed to deployer
- remove the nonce
- remove from, deployer pays for the deploy

### JIRA ticket:
https://rchain.atlassian.net/browse/RCHAIN-3132

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: node/src/main/scala/coop/rchain/node/Main.scala [hunk 1]
```diff
@@ -65,13 +65,11 @@ object Main {
       case Eval(files) => new ReplRuntime().evalProgram[Task](files)
       case Repl        => new ReplRuntime().replProgram[Task].as(())
       case Diagnostics => diagnostics.client.Runtime.diagnosticsProgram[Task]
-      case Deploy(address, phlo, phloPrice, nonce, validAfterBlock, userId, location) =>
+      case Deploy(phlo, phloPrice, validAfterBlock, userId, location) =>
```
Comment (KentShikama) [inline node/src/main/scala/coop/rchain/node/Main.scala]: deployer?

File: node/src/main/scala/coop/rchain/node/configuration/Configuration.scala [hunk 2]
```diff
@@ -187,10 +187,8 @@ object Configuration {
         //TODO: change the defaults before main net
         import options.deploy._
         Deploy(
-          from.getOrElse("0x"),
           phloLimit(),
           phloPrice(),
-          nonce.getOrElse(0),
```
Comment (KentShikama) [inline node/src/main/scala/coop/rchain/node/configuration/Configuration.scala]: two lines below this

File: node/src/main/scala/coop/rchain/node/configuration/commandline/Options.scala [hunk 3]
```diff
@@ -379,10 +374,6 @@ final case class Options(arguments: Seq[String]) extends ScallopConf(arguments)
         "Set this value to one less than the current block height: you have 50 blocks to get this transaction into the chain."
     )
 
-    val nonce = opt[Int](
-      descr = "This allows you to overwrite your own pending transactions that use the same nonce."
-    )
-
```
Comment (KentShikama) [inline node/src/main/scala/coop/rchain/node/configuration/commandline/Options.scala]: Should userId below be deployer?

File: node/src/main/scala/coop/rchain/node/Main.scala (hunk 1 above)
Comment (odeac) [reply node/src/main/scala/coop/rchain/node/Main.scala]: done

File: node/src/main/scala/coop/rchain/node/configuration/commandline/Options.scala (hunk 3 above)
Comment (odeac) [reply node/src/main/scala/coop/rchain/node/configuration/commandline/Options.scala]: fixed

File: casper/src/main/scala/coop/rchain/casper/util/ProtoUtil.scala [hunk 4]
```diff
@@ -531,7 +531,7 @@ object ProtoUtil {
 
   def getRholangDeployParams(dd: DeployData): (Par, Par, Par, Par) = {
     val phloPrice: Par = Par(exprs = Seq(Expr(Expr.ExprInstance.GInt(dd.phloPrice))))
-    val userId: Par    = Par(exprs = Seq(Expr(Expr.ExprInstance.GByteArray(dd.user))))
+    val userId: Par    = Par(exprs = Seq(Expr(Expr.ExprInstance.GByteArray(dd.deployer))))
```
Comment (ArturGajowy) [inline casper/src/main/scala/coop/rchain/casper/util/ProtoUtil.scala]: we should probably text-search for occurenceso of userId and replace those too

**Pull request merged**